### PR TITLE
upgraded xchain-thorchain

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
Upgraded:
- xchain-thorchain
 - xchainjs/xchain-util

Added:
- xchain-litecoin packages

Fixed:
- new introduced chains fixs

### CODE REVIEW

File: src/renderer/helpers/chainHelper.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ export const getChainAsset = (chain: Chain): Asset => {
       return AssetAtom
     case 'POLKA':
       return AssetDOT
+    case 'BCH': {
+      // @todo add BCH support
+      return AssetBTC
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/helpers/chainHelper.ts:23]: as bch stuff is not completed yet at the `xchain-*` im not sure what is the best way to set default here. suggestions are welcome

File: src/renderer/helpers/addressHelper.ts [hunk 2]
```diff
@@ -26,5 +37,12 @@ export const getAddressPrefixLength = (chain: Chain, network: string): number =>
       return getPolkadotPrefix(network).length
     case THORChain:
       return getThorchainPrefix(network).length
+    case LTCChain: {
+      return getLitecoinPrefix(network).length
+    }
+    // @todo: support missed chain "BCH"
+    case BCHChain: {
+      return 0
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/helpers/addressHelper.ts:45]: `0` is fine for now

File: src/renderer/helpers/addressHelper.ts [hunk 3]
```diff
@@ -26,5 +37,12 @@ export const getAddressPrefixLength = (chain: Chain, network: string): number =>
       return getPolkadotPrefix(network).length
     case THORChain:
       return getThorchainPrefix(network).length
+    case LTCChain: {
+      return getLitecoinPrefix(network).length
+    }
+    // @todo: support missed chain "BCH"
```
Comment (veado) [inline src/renderer/helpers/addressHelper.ts:43]: TODO (@thatStrangeGuyThorchain or @asgdx-team) - see [Epic] Support Bitcoin Cash https://github.com/thorchain/asgardex-electron/issues/821

File: src/renderer/helpers/chainHelper.ts [hunk 4]
```diff
@@ -18,6 +18,13 @@ export const getChainAsset = (chain: Chain): Asset => {
       return AssetAtom
     case 'POLKA':
       return AssetDOT
+    case 'BCH': {
+      // @todo add BCH support
```
Comment (veado) [inline src/renderer/helpers/chainHelper.ts:22]: TODO (@thatStrangeGuyThorchain or @asgdx-team) - see [Epic] Support Bitcoin Cash https://github.com/thorchain/asgardex-electron/issues/821

File: src/renderer/helpers/chainHelper.ts (hunk 1 above)
Comment (veado) [reply src/renderer/helpers/chainHelper.ts:23]: `{ chain: BCHChain, symbol: 'BCH', ticker: 'BCH' }` + add a comment to link it to https://github.com/xchainjs/xchainjs-lib/issues/206

File: src/renderer/services/chain/fees/deposit.ts [hunk 5]
```diff
@@ -102,6 +102,10 @@ const depositFeeByChain$ = (chain: Chain, type: DepositType): FeeLD => {
       return Rx.of(RD.failure(Error('Deposit fee for Cosmos has not been implemented')))
     case 'POLKA':
       return Rx.of(RD.failure(Error('Deposit fee for Polkadot has not been implemented')))
+    case 'BCH':
+      return Rx.of(RD.failure(Error('Deposit fee for Bitcoincash has not been implemented')))
```
Comment (veado) [inline src/renderer/services/chain/fees/deposit.ts:106]: [nitpick] `Bitcoin Cash`

File: src/renderer/services/chain/fees/withdraw.ts [hunk 6]
```diff
@@ -42,6 +42,10 @@ const withdrawFeeByChain$ = (chain: Chain): FeeLD => {
       return Rx.of(RD.failure(Error(`Deposit fee for Cosmos has not been implemented`)))
     case 'POLKA':
       return Rx.of(RD.failure(Error(`Deposit fee for Polkadot has not been implemented`)))
+    case 'BCH':
+      return Rx.of(RD.failure(Error(`Deposit fee for Bitcoincash has not been implemented`)))
```
Comment (veado) [inline src/renderer/services/chain/fees/withdraw.ts:46]: [nitpick] `Bitcoin Cash`

File: src/renderer/services/chain/transaction/common.ts [hunk 7]
```diff
@@ -54,6 +64,12 @@ export const sendTx$ = ({ asset, recipient, amount, memo, txType, feeOptionKey }
     case PolkadotChain:
       // not available yet
       return txFailure$(`Tx stuff has not been implemented for Polkadot yet`)
+    case BCHChain:
+      // not available yet
+      return txFailure$(`Tx stuff has not been implemented for Bitcoincash yet`)
```
Comment (veado) [inline src/renderer/services/chain/transaction/common.ts:69]: [nitpick] `Bitcoin Cash`

File: src/renderer/services/chain/transaction/common.ts [hunk 8]
```diff
@@ -80,5 +96,9 @@ export const txStatusByChain$ = (txHash: TxHash, chain: Chain): TxLD => {
       return failure$(`txStatus$ has not been implemented for Cosmos`)
     case PolkadotChain:
       return failure$(`txStatus$ has not been implemented for Polkadot`)
+    case BCHChain:
+      return failure$(`txStatus$ has not been implemented for Bitcoincash`)
```
Comment (veado) [inline src/renderer/services/chain/transaction/common.ts:100]: [nitpick] `Bitcoin Cash`

File: src/renderer/services/clients/fees.ts [hunk 9]
```diff
@@ -30,6 +41,11 @@ const getDefaultFeesByChain = (chain: Chain) => {
     case PolkadotChain:
       // TODO @Veado: Handle network
       return Polkadot.getDefaultFees('testnet')
+    case BCHChain:
+      // @todo handle BCH fees
+      return THOR.getDefaultFees()
```
Comment (veado) [inline src/renderer/services/clients/fees.ts:46]:
```
return {
    type: 'byte',
    fast: ZERO_BASE_AMOUNT,
    fastest: ZERO_BASE_AMOUNT,
    average: ZERO_BASE_AMOUNT,
  }
```

File: src/renderer/services/clients/fees.ts [hunk 10]
```diff
@@ -30,6 +41,11 @@ const getDefaultFeesByChain = (chain: Chain) => {
     case PolkadotChain:
       // TODO @Veado: Handle network
       return Polkadot.getDefaultFees('testnet')
+    case BCHChain:
+      // @todo handle BCH fees
```
Comment (veado) [inline src/renderer/services/clients/fees.ts:45]: TODO (@thatStrangeGuyThorchain or @asgdx-team) - see [Epic] Support Bitcoin Cash https://github.com/thorchain/asgardex-electron/issues/821

**Review by veado — CHANGES_REQUESTED** [review]

**Review by veado — DISMISSED** [review]

**Review by veado — APPROVED** [review]

**Pull request merged**